### DISCUSSION TITLE
How to calculate speed?

### DISCUSSION THREAD

**Discussion by adrian-brennan** [zepp-health/zeppos-forum#292] [Q&A] [answered]
I'm thinking about a new app idea, and want to detect the speed that the device is moving. As far as I can see, there's no "speed" sensor.

I could calculate speed by e.g. every few seconds getting latitude/longitude values from the [Geolocation API](https://docs.zepp.com/docs/reference/device-app-api/newAPI/sensor/Geolocation/) and doing the math to figure out the distance moved. Is there an easier approach to this?

Thanks :)

**Comment by silver-zepp** [comment, marked as answer]
Define the "speed" sensor :)

You have a valid approach using a GPS for that task. The second option would be to use a cadence sensor. They are natively supported in our Cycling Workouts but if you're making your own app, then you would rely on the [BLE Master](https://github.com/silver-zepp/zeppos-easy-ble) which is currently supported only on the Balance watch. That said these sensors are still sort of "invasive" as they rely on the rotational force compared to GPS.

**Reply by adrian-brennan** [reply]
Got it - thanks for the swift feedback
